typescript/vscode-ext/packages/language-server/src/server.ts

```diff
@@ -348,20 +348,33 @@ export function startServer(options?: LSOptions): void {
   }
 
   connection.onDefinition((params: DeclarationParams) => {
-    return undefined
-    // const doc = getDocument(params.textDocument.uri)
-    // if (doc) {
-    //   const db = bamlCache.getFileCache(doc)
-    //   if (db) {
-    //     return MessageHandler.handleDefinitionRequest(db, doc, params)
-    //   } else if (doc.languageId === 'python') {
-    //     const db = bamlCache.lastBamlDir?.cache
-    //     console.log(` python: ${doc.uri} files: ${db?.getDocuments().length}`)
-    //     if (db) {
-    //       return MessageHandler.handleDefinitionRequest(db, doc, params)
-    //     }
-    //   }
-    // }
+    
+    
+    
+    //accesses document from uri
+    const doc = getDocument(params.textDocument.uri)
+    
+    if (doc) {
+      
+      //accesses project from uri via bamlProjectManager
+      const proj = bamlProjectManager.getProjectById((URI.parse(doc.uri)))
+      if (proj) {
+
+        //returns the definition of reference within the project
+        return proj.handleDefinitionRequest(doc, params.position)
+
+      } 
+  
+
+      
+      else if (doc.languageId === 'python') {
```

Comment:
i dont think we need to do anything for python language -- or what is this supposed to do?

Reply:
It was already there, so I didn't modify it.

Reply:
we can delete this python-specific code for now. The Since we're going from commenting it out to enabling it, and we don't know what it is supposed to do.